### PR TITLE
treewide: use https addresses where possible

### PR DESCRIPTION
**Pull request by iblech**
###### Motivation for this change

In my social circle, it's considered slightly inpolite to use http instead of https addresses. :-) Therefore I thought to give rewriting of all http addresses in the nixpkgs repository a go, just as an experiment.

Of the approximately 13000 http links contained in this repository, 5000 could be rewritten as https links, 3000 couldn't be upgraded, and 4000 were 404's. The numbers are not very precise, since I didn't bother to accurately filter local and example addresses, and also because at one point during the test my internet connection was flaky.

If you want to study the diff, then I recommend using `git diff -U0` to only show changed lines. The context, which is usually quite helpful, is here distracting.

This pull request isn't primarily meant to be merged. In particular, I didn't rebuild all of NixOS to check for breakage. :-) Rather, I just wanted to raise awareness of this issue. (I'm not trying to start a religious war and don't particularly care. Of course the issue is not really an issue for NixOS since we verify sha256sums. Still we could gain street creds if we could proclaim that "NixOS doesn't contain a single http link".)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by iblech** [comment]
For anyone wondering, the Travis build fails with "argument list too long" because the number of files touched by this commit is too large. This is probably a good thing.

File: doc/coding-conventions.xml [hunk 1]
```diff
@@ -670,7 +670,7 @@ src = fetchFromGitHub {
 patches = [
   (fetchpatch {
     name = "fix-check-for-using-shared-freetype-lib.patch";
-    url = "http://git.ghostscript.com/?p=ghostpdl.git;a=patch;h=8f5d285";
+    url = "https://git.ghostscript.com/?p=ghostpdl.git;a=patch;h=8f5d285";
```
Comment (grahamc) [inline doc/coding-conventions.xml]: 👍

File: doc/introduction.md [hunk 2]
```diff
@@ -7,26 +7,26 @@ date: 2015-11-25
 # Introduction
 
 The Nix Packages collection (Nixpkgs) is a set of thousands of packages for the
-[Nix package manager](http://nixos.org/nix/), released under a
+[Nix package manager](https://nixos.org/nix/), released under a
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/introduction.md [hunk 3]
```diff
@@ -7,26 +7,26 @@ date: 2015-11-25
...
 [permissive MIT/X11 license](https://github.com/NixOS/nixpkgs/blob/master/COPYING).
 Packages are available for several platforms, and can be used with the Nix
 package manager on most GNU/Linux distributions as well as NixOS.
 
 This manual primarily describes how to write packages for the Nix Packages collection
 (Nixpkgs). Thus it’s mainly for packagers and developers who want to add packages to
 Nixpkgs. If you like to learn more about the Nix package manager and the Nix
-expression language, then you are kindly referred to the [Nix manual](http://nixos.org/nix/manual/).
+expression language, then you are kindly referred to the [Nix manual](https://nixos.org/nix/manual/).
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/introduction.md [hunk 4]
```diff
@@ -7,26 +7,26 @@ date: 2015-11-25
...
 
 ## Overview of Nixpkgs
 
 Nix expressions describe how to build packages from source and are collected in
 the [nixpkgs repository](https://github.com/NixOS/nixpkgs). Also included in the
 collection are Nix expressions for
-[NixOS modules](http://nixos.org/nixos/manual/index.html#sec-writing-modules).
+[NixOS modules](https://nixos.org/nixos/manual/index.html#sec-writing-modules).
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/introduction.md [hunk 5]
```diff
@@ -7,26 +7,26 @@ date: 2015-11-25
...
 With these expressions the Nix package manager can build binary packages.
 
 Packages, including the Nix packages collection, are distributed through
-[channels](http://nixos.org/nix/manual/#sec-channels). The collection is
+[channels](https://nixos.org/nix/manual/#sec-channels). The collection is
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/introduction.md [hunk 6]
```diff
@@ -36,12 +36,12 @@ security updates. More up to date packages and modules are available via the
 
 Both `nixos-unstable` and `nixpkgs` follow the `master` branch of the Nixpkgs
 repository, although both do lag the `master` branch by generally
-[a couple of days](http://howoldis.herokuapp.com/). Updates to a channel are
+[a couple of days](https://howoldis.herokuapp.com/). Updates to a channel are
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/introduction.md [hunk 7]
```diff
@@ -36,12 +36,12 @@ security updates. More up to date packages and modules are available via the
...
 distributed as soon as all tests for that channel pass, e.g.
-[this table](http://hydra.nixos.org/job/nixpkgs/trunk/unstable#tabs-constituents)
+[this table](https://hydra.nixos.org/job/nixpkgs/trunk/unstable#tabs-constituents)
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/introduction.md [hunk 8]
```diff
@@ -36,12 +36,12 @@ security updates. More up to date packages and modules are available via the
...
 shows the status of tests for the `nixpkgs` channel.
 
-The tests are conducted by a cluster called [Hydra](http://nixos.org/hydra/),
+The tests are conducted by a cluster called [Hydra](https://nixos.org/hydra/),
```
Comment (grahamc) [inline doc/introduction.md]: 👍

File: doc/languages-frameworks/bower.xml [hunk 9]
```diff
@@ -5,7 +5,7 @@
 <title>Bower</title>
 
 <para>
-  <link xlink:href="http://bower.io">Bower</link> is a package manager
+  <link xlink:href="https://bower.io">Bower</link> is a package manager
```
Comment (grahamc) [inline doc/languages-frameworks/bower.xml]: 👍

File: doc/languages-frameworks/haskell.md [hunk 10]
```diff
@@ -9,7 +9,7 @@ date: 2015-06-01
 ## How to install Haskell packages
 
 Nixpkgs distributes build instructions for all Haskell packages registered on
-[Hackage](http://hackage.haskell.org/), but strangely enough normal Nix package
+[Hackage](https://hackage.haskell.org/), but strangely enough normal Nix package
```
Comment (grahamc) [inline doc/languages-frameworks/haskell.md]: 👍

File: doc/languages-frameworks/haskell.md [hunk 11]
```diff
@@ -249,7 +249,7 @@ of the wrapper script then executes the real `ghc`, but passes the path to the
 new `lib` directory using GHC's `-B` flag.
 
 The purpose of those environment variables is to work around an impurity in the
-popular [ghc-paths](http://hackage.haskell.org/package/ghc-paths) library. That
+popular [ghc-paths](https://hackage.haskell.org/package/ghc-paths) library. That
```
Comment (grahamc) [inline doc/languages-frameworks/haskell.md]: 👍

File: doc/languages-frameworks/haskell.md [hunk 12]
```diff
@@ -345,13 +345,13 @@ workarounds.
 
 ### How to build a Haskell project using Stack
 
-[Stack](http://haskellstack.org) is a popular build tool for Haskell projects.
+[Stack](https://haskellstack.org) is a popular build tool for Haskell projects.
```
Comment (grahamc) [inline doc/languages-frameworks/haskell.md]: 👍

File: doc/languages-frameworks/haskell.md [hunk 13]
```diff
@@ -345,13 +345,13 @@ workarounds.
...
 It has first-class support for Nix. Stack can optionally use Nix to
 automatically select the right version of GHC and other build tools to build,
 test and execute apps in an existing project downloaded from somewhere on the
 Internet. Pass the `--nix` flag to any `stack` command to do so, e.g.
 ```shell
-git clone --recursive http://github.com/yesodweb/wai
+git clone --recursive https://github.com/yesodweb/wai
```
Comment (grahamc) [inline doc/languages-frameworks/haskell.md]: 👍

File: doc/languages-frameworks/haskell.md [hunk 14]
```diff
@@ -890,7 +890,7 @@ failures. Additionally the source of your package is gotten from first invoking
 #### checkUnusedPackages
 
 Applying `haskell.lib.checkUnusedPackages` to a Haskell package invokes
-the [packunused](http://hackage.haskell.org/package/packunused) tool on the
+the [packunused](https://hackage.haskell.org/package/packunused) tool on the
```
Comment (grahamc) [inline doc/languages-frameworks/haskell.md]: this file looks good

File: doc/languages-frameworks/python.md [hunk 15]
```diff
@@ -563,7 +563,7 @@ The following is an example:
     propagatedBuildInputs = [ self.ZopeInterface ];
 
     meta = {
-      homepage = http://twistedmatrix.com/;
+      homepage = https://twistedmatrix.com/;
       description = "Twisted, an event-driven networking engine written in Python";
       license = stdenv.lib.licenses.mit;
     };
```
Comment (grahamc) [inline doc/languages-frameworks/python.md]: This file looks good.

File: doc/languages-frameworks/ruby.xml [hunk 16]
```diff
@@ -34,7 +34,7 @@ bundlerEnv rec {
 
   meta = with lib; {
     description = "A monitoring framework that aims to be simple, malleable, and scalable";
-    homepage    = http://sensuapp.org/;
+    homepage    = https://sensuapp.org/;
     license     = with licenses; mit;
     maintainers = with maintainers; [ theuni ];
     platforms   = platforms.unix;
```
Comment (grahamc) [inline doc/languages-frameworks/ruby.xml]: 👍

File: doc/meta.xml [hunk 17]
```diff
@@ -122,13 +122,13 @@ meta-attributes</title>
...
   </varlistentry>
 
   <varlistentry>
     <term><varname>downloadPage</varname></term>
     <listitem><para>The page where a link to the current version can be found.  Example:
-    <literal>http://ftp.gnu.org/gnu/hello/</literal></para></listitem>
+    <literal>https://ftp.gnu.org/gnu/hello/</literal></para></listitem>
   </varlistentry>
 
   <varlistentry>
```
Comment (grahamc) [inline doc/meta.xml]: this file looks good.

**Review by grahamc — COMMENTED** [review]
I reviewed some. how did you test it? A script?

File: pkgs/applications/editors/eclipse/build-eclipse.nix [hunk 18]
```diff
@@ -53,7 +53,7 @@ stdenv.mkDerivation rec {
   ''; # */
 
   meta = {
-    homepage = http://www.eclipse.org/;
+    homepage = https://www.eclipse.org/;
```
Comment (rycee) [inline pkgs/applications/editors/eclipse/build-eclipse.nix]: 👍

File: pkgs/development/eclipse/ecj/default.nix [hunk 19]
```diff
@@ -51,7 +51,7 @@ stdenv.mkDerivation rec {
 
     homepage = http://www.eclipse.org/jdt/core/index.php;
 
-    # http://www.eclipse.org/legal/epl-v10.html (free software, copyleft)
+    # https://www.eclipse.org/legal/epl-v10.html (free software, copyleft)
```
Comment (rycee) [inline pkgs/development/eclipse/ecj/default.nix]: 👍

**Review by rycee — COMMENTED** [review]
I like the idea! The Eclipse and Perl related links look OK.

**Comment by FRidh** [comment]
I like the idea, but this is too large to review/test. In this case I think it is important to share the script that was used for making and testing this change. As it is now also outdated I am closing this.

**Pull request closed without merging**